packages/flutter/lib/src/material/dialog.dart

```diff
@@ -1129,3 +1140,71 @@ double _paddingScaleFactor(double textScaleFactor) {
   // a non-scaled padding of 24 will produce a padding between 24 and 8.
   return lerpDouble(1.0, 1.0 / 3.0, clampedTextScaleFactor - 1.0)!;
 }
+
+// Generate a DialogTheme that represents the M2 default values.
+// This was generated by hand from the previous hand coded defaults
+// for M2. It uses get method overrides instead of properties to
+// avoid computing values that we may not need upfront.
+class _M2Defaults extends DialogTheme {
+  _M2Defaults(this.context)
+    : _textTheme = Theme.of(context).textTheme;
+
+  final BuildContext context;
+  final TextTheme _textTheme;
+
+  @override
+  AlignmentGeometry? get alignment =>  Alignment.center;
+
+  @override
+  Color? get backgroundColor => Theme.of(context).dialogBackgroundColor;
+
+  @override
+  double? get elevation => 24.0;
+
+  @override
+  ShapeBorder? get shape => const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(4.0)));
```

Comment:
Same with these constant properties, just make them parameters to the super class in the constructor.